Framework/PythonInterface/mantid/kernel/src/Environment/ErrorHandling.cpp

```diff
@@ -64,8 +70,7 @@ void throwRuntimeError(const bool withTrace) {
   Py_XDECREF(traceback);
   Py_XDECREF(exception);
   Py_XDECREF(value);
-  // Raise this error as a C++ error
-  throw std::runtime_error(msg.str());
+  m_msg = builder.str();
```

Comment:
Just out of curiosity, is this constructor marked as `noexcept` just for performance reasons, or was it required for some other reason? I get the feeling that exceptions could be raised from here, producing an abrupt termination.